Dependency version conflict between Dolphin and EM

@gyeongin has found this error while tackling #70. When Dolphin has a dependency to EM, a conflict occurs while launching Dolphin, which consequently causes RuntimeException.

Below is the `driver.stderr` from Dolphin:

```
…
SLF4J: The requested version 1.6.99 by your slf4j binding is not compatible with [1.5.5, 1.5.6, 1.5.7, 1.5.8]
SLF4J: See http://www.slf4j.org/codes.html#version_mismatch for further details.
…
SLF4J: The requested version 1.6.99 by your slf4j binding is not compatible with [1.5.5, 1.5.6, 1.5.7, 1.5.8]
SLF4J: See http://www.slf4j.org/codes.html#version_mismatch for further details.
2015-08-20 14:00:37,092 SEVERE reef.runtime.common.Launcher.fail main | Unable to run LaunchClass
java.lang.RuntimeException: org.apache.reef.tang.exceptions.InjectionException: Could not invoke constructor: new                          DefaultRemoteManagerImplementation(String ManagerName = REEF_LAUNCHER, String HostAddress = ##UNKNOWN##, Integer Port = 0, MessageCodec =  new REEFMessageCodec(), ErrorHandler = [ClassNodeImpl 'org.apache.reef.runtime.common.launch.REEFErrorHandler']: org.apache.reef.runtime.  common.launch.REEFErrorHandler(org.apache.reef.runtime.common.utils.RemoteManager,java.lang.String org.apache.reef.runtime.common.launch.  parameters.ErrorHandlerRID,java.lang.String org.apache.reef.runtime.common.launch.parameters.LaunchID,org.apache.reef.runtime.common.utils.ExceptionCodec),  = REEFErrorHandler{remoteManager=org.apache.reef.tang.InjectionFuture@7db1be6, launchID='Logistic Regression',           errorHandlerRID='socket://192.168.0.102:17777'}, Boolean OrderingGuarantee = true, Integer NumberOfTries = 3, Integer RetryTimeout =       10000, LocalAddressProvider = new HostnameBasedLocalAddressProvider(), TcpPortProvider = new RangeTcpPortProvider(Integer                  TcpPortRangeBegin = 10000, Integer TcpPortRangeCount = 10000, Integer TcpPortRangeTryCount = 1000))
  at org.apache.reef.tang.InjectionFuture.get(InjectionFuture.java:127)
  at org.apache.reef.tang.implementation.java.InjectorImpl.getInstance(InjectorImpl.java:519)
  at org.apache.reef.tang.implementation.java.InjectorImpl.getInstance(InjectorImpl.java:531)
  at org.apache.reef.runtime.common.Launcher.main(Launcher.java:109)
Caused by: org.apache.reef.tang.exceptions.InjectionException: Could not invoke constructor: new DefaultRemoteManagerImplementation(String ManagerName = REEF_LAUNCHER, String HostAddress = ##UNKNOWN##, Integer Port = 0, MessageCodec = new REEFMessageCodec(), ErrorHandler =     [ClassNodeImpl 'org.apache.reef.runtime.common.launch.REEFErrorHandler']: org.apache.reef.runtime.common.launch.REEFErrorHandler(org.      apache.reef.runtime.common.utils.RemoteManager,java.lang.String org.apache.reef.runtime.common.launch.parameters.ErrorHandlerRID,java.lang.String org.apache.reef.runtime.common.launch.parameters.LaunchID,org.apache.reef.runtime.common.utils.ExceptionCodec),  =                  REEFErrorHandler{remoteManager=org.apache.reef.tang.InjectionFuture@7db1be6, launchID='Logistic Regression', errorHandlerRID='socket://192.168.0.102:17777'}, Boolean OrderingGuarantee = true, Integer NumberOfTries = 3, Integer RetryTimeout = 10000, LocalAddressProvider = new   HostnameBasedLocalAddressProvider(), TcpPortProvider = new RangeTcpPortProvider(Integer TcpPortRangeBegin = 10000, Integer                 TcpPortRangeCount = 10000, Integer TcpPortRangeTryCount = 1000))
  at org.apache.reef.tang.implementation.java.InjectorImpl.injectFromPlan(InjectorImpl.java:651)
  at org.apache.reef.tang.implementation.java.InjectorImpl.injectFromPlan(InjectorImpl.java:657)
  at org.apache.reef.tang.implementation.java.InjectorImpl.injectFromPlan(InjectorImpl.java:623)
  at org.apache.reef.tang.implementation.java.InjectorImpl.getInstance(InjectorImpl.java:513)
  at org.apache.reef.tang.implementation.java.InjectorImpl.getInstance(InjectorImpl.java:531)
  at org.apache.reef.tang.InjectionFuture.get(InjectionFuture.java:118)
  ... 3 more
Caused by: java.lang.NoSuchMethodError: org.slf4j.helpers.MessageFormatter.format(Ljava/lang/String;Ljava/lang/Object;)Lorg/slf4j/helpers/ FormattingTuple;
  at org.slf4j.impl.Log4jLoggerAdapter.debug(Log4jLoggerAdapter.java:228)
  at io.netty.util.internal.logging.Slf4JLogger.debug(Slf4JLogger.java:76)
  at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:75)
  at io.netty.util.internal.PlatformDependent.getSystemClassLoader(PlatformDependent.java:405)
  at io.netty.util.internal.PlatformDependent.isAndroid0(PlatformDependent.java:411)
  at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:61)
  at io.netty.util.Signal.<clinit>(Signal.java:31)
  at io.netty.util.concurrent.DefaultPromise.<clinit>(DefaultPromise.java:39)
  at io.netty.util.concurrent.GlobalEventExecutor.<init>(GlobalEventExecutor.java:48)
  at io.netty.util.concurrent.GlobalEventExecutor.<clinit>(GlobalEventExecutor.java:44)
  at org.apache.reef.wake.remote.transport.netty.NettyMessagingTransport.<init>(NettyMessagingTransport.java:80)
  at org.apache.reef.wake.remote.impl.DefaultRemoteManagerImplementation.<init>(DefaultRemoteManagerImplementation.java:117)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
  at org.apache.reef.tang.implementation.java.InjectorImpl.injectFromPlan(InjectorImpl.java:635)
  ... 8 more
...
```

The conflict may be resolved specifying or ignoring the version of slf4j in the `pom.xml`.

--- a/pom.xml
+++ b/pom.xml
@@ -134,6 +134,11 @@
         <scope>test</scope>
       </dependency>
       <dependency>
+        <groupId>org.slf4j</groupId>
+        <artifactId>slf4j-api</artifactId>
+        <version>1.7.5</version>
+      </dependency>
+      <dependency>
         <groupId>org.apache.mahout</groupId>
         <artifactId>mahout-math</artifactId>
         <version>${mahout.version}</version>

